### ISSUE TITLE
Indexes backward compatibility broken in 3.5+

### ISSUE THREAD

**Issue by soualid** [arangodb/arangodb#11275]
Hello,

Using a single server database, it seems that since the 3.5 series, an unique index set on a subfield of a document could lead to a crash during updates if this field has a `null` value stored in the database. 

See the repro below, which is reproducible using a fresh arandodb 3.5.4 docker image (haven't tested this on the 3.6 serie yet).

```
db._drop('repro')
db._createDocumentCollection('repro')
db.repro.ensureIndex({fields:['nativeIdentity.emailAddress'], type: 'hash', unique: true})
db.repro.insert([
  {
    "nativeIdentity": null
  }
])
db._query('for r in repro update r with { anotherField: "test2" } in repro')
```

Will return the following error:

```
JavaScript exception in file '/usr/share/arangodb3/js/client/modules/@arangodb/arangosh.js' at 99,7: ArangoError 4: Expecting Object (while executing) (exception location: /work/ArangoDB/arangod/RestHandler/RestCursorHandler.cpp:283). Please report this error to arangodb.com
!      throw error;
!      ^
stacktrace: ArangoError: Expecting Object (while executing) (exception location: /work/ArangoDB/arangod/RestHandler/RestCursorHandler.cpp:283). Please report this error to arangodb.com
    at Object.exports.checkRequestResult (/usr/share/arangodb3/js/client/modules/@arangodb/arangosh.js:97:21)
    at ArangoStatement.execute (/usr/share/arangodb3/js/client/modules/@arangodb/arango-statement.js:172:12)
    at Proxy.ArangoDatabase._query (/usr/share/arangodb3/js/client/modules/@arangodb/arango-database.js:954:45)
```

This is breaking backward compatibility because such updates worked in previous versions and consequently, making the update process a bit risky if we have to analyze all the datas stored in the database to avoid this case.

Thank you for your feedbacks on this!

**Comment by jsteemann** [comment]
Just verified it and can confirm that there is this behavior change in 3.5. And that 3.6 and 3.7 are also affected by this issue. 3.4 is not.

**Comment by maxkernbach** [comment]
Hi @soualid,

Version 3.5.5, which includes a fix for this issue, has been released.

From the [Changelog](https://github.com/arangodb/arangodb/blob/69a307b8c03aa23abbac7a9a65a21001b6ea4fce/CHANGELOG#L39):

> 
> * Fixed issue #11275: indexes backward compatibility broken in 3.5+.

Please find the latest release in the [download section](https://www.arangodb.com/download/).
